tests/adv_widgets_test.py

```diff
@@ -0,0 +1,92 @@
+# Advanced widgets test cases.
+#
+# Copyright (C) 2017  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+import unittest
+from unittest.mock import patch
+
+from io import StringIO
+
+from tests import UtilityMixin
+
+from simpleline.render.adv_widgets import GetInputScreen, GetPassInputScreen
+
+
+@patch('simpleline.render.io_manager.InOutManager._get_input')
+@patch('sys.stdout', new_callable=StringIO)
+class AdvWidgets_TestCase(unittest.TestCase, UtilityMixin):
+    def setUp(self):
+        self.correct_input = False
+
+    def test_GetTextInput(self, stdout_mock, stdin_mock):
```

Comment:
This method name is terrifying, use `test_get_text_input`, please :-)

Reply:
It's name of the class not test for getting text input.

Reply:
Fixed.